### PR TITLE
only show flavor above 0.1u

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
if a pill has say 0.01u carbon you could taste it
this pr fixes and only lets you taste it if its above 0.1u in the solution being ingested

**Media**
less than 0.1u sugar

does not taste sweet!

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
Can this be configurable so that certain flavors have a higher or lower threshold for being noticed? Something like oil should be immediately tasted whereas some other flavors might be more subtle.

**Comment by deltanedas** [comment]
ok ill add thing to reagent prototype

File: Content.Server/Nutrition/EntitySystems/FlavorProfileSystem.cs [hunk 1]
```diff
@@ -25,12 +25,12 @@ public sealed class FlavorProfileSystem : EntitySystem
...
         FlavorProfileComponent? flavorProfile = null)
     {
-        var flavors = new HashSet<string>();
         if (!Resolve(uid, ref flavorProfile, false))
         {
             return Loc.GetString(BackupFlavorMessage);
         }
 
+        var flavors = new HashSet<string>();
         flavors.UnionWith(flavorProfile.Flavors);
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/EntitySystems/FlavorProfileSystem.cs:34]: Can you pass the hashset into the ctor or is that just lists? At a minimum should initialize the capacity. to flavorProfile.Flavors.Count.

File: Content.Shared/Chemistry/Reagent/ReagentPrototype.cs [hunk 2]
```diff
@@ -63,6 +63,12 @@ public sealed class ReagentPrototype : IPrototype, IInheritingPrototype
         [DataField("flavor")]
         public string Flavor { get; } = default!;
 
+        /// <summary>
+        /// There must be at least this much quantity in a solution to be tasted.
+        /// </summary>
+        [DataField("flavorMinimum")]
+        public float FlavorMinimum = 0.1f;
```
Comment (metalgearsloth) [inline Content.Shared/Chemistry/Reagent/ReagentPrototype.cs:70]: Should this be fixedpoint2?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/Chemistry/Reagent/ReagentPrototype.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Shared/Chemistry/Reagent/ReagentPrototype.cs:70]: probably

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**